### PR TITLE
Flutter views can gain focus

### PR DESCRIPTION
**Pull request by tugorez**
I am unsure why the `tabindex` was removed when semantics were enabled. It seems this change was made without a clear explanation (by me). This PR shouldn't cause any issues as Flutter Views already have a tabindex, we're not adding a new one. This change is necessary because the semantics text strategy refocuses the view on deactivation, requiring the Flutter view to be focusable.

ThIs PR is a requirement to enable https://github.com/flutter/engine/pull/54966. 

https://github.com/flutter/flutter/issues/153022

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by ditman — APPROVED** [review]
This PR seems correct. What it does is to make the current flutter view focusable (so it can retain the focus if needed), but inaccessible by keyboard. See [`tabindex` docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex#:~:text=A%20negative%20value%20(the%20exact%20negative%20value%20doesn%27t%20actually%20matter%2C%20usually%20tabindex%3D%22%2D1%22)%20means%20that%20the%20element%20is%20not%20reachable%20via%20sequential%20keyboard%20navigation.).

**Comment by ditman** [comment]
I'm taking a quick stab at attaching the focus handlers to the flutter view root, rather than the body.

**Comment by ditman** [comment]
Looks like I broke something in Safari with my latest changes, reviewing:

* https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8731518283441607553/+/u/test:_run_suite_chrome-dart2js-html-engine/stdout

File: lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart [hunk 1]
```diff
@@ -127,29 +130,32 @@ final class ViewFocusBinding {
...
-    } else {
-      // A tabindex with value zero means the DOM element can be reached by using
-      // the keyboard (tab, shift + tab). When its value is -1 it is still focusable
-      // but can't be focused by the result of keyboard events This is specially
-      // important when the semantics tree is enabled as it puts DOM nodes inside
-      // the flutter view and having it with a zero tabindex messes the focus
-      // traversal order when pressing tab or shift tab.
-      rootElement?.setAttribute('tabindex', reachableByKeyboard ? 0 : -1);
-    }
+    // A tabindex with value zero means the DOM element can be reached by using
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart:152]: nit: remove the "by"

File: lib/web_ui/test/engine/platform_dispatcher/view_focus_binding_test.dart [hunk 2]
```diff
@@ -68,28 +68,6 @@ void testMain() {
       expect(view2.dom.rootElement.getAttribute('tabindex'), '0');
     });
 
-    test('never marks the views as focusable with semantincs enabled', () async {
```
Comment (yjbanov) [inline lib/web_ui/test/engine/platform_dispatcher/view_focus_binding_test.dart:71]: Should we also _add_ some tests? :)

Comment (ditman) [reply lib/web_ui/test/engine/platform_dispatcher/view_focus_binding_test.dart:71]: Well, there's tests already (this file is just showing the removed test, but there's others to verify the rest of the functionality!)

**Review by yjbanov — APPROVED** [review]
lgtm

**Comment by ditman** [comment]
One last update and applying autosubmit.

**Pull request merged**